lib/Open/directory.js

```diff
@@ -163,16 +163,24 @@ module.exports = function centralDirectory(source, options) {
             if (extractPath.indexOf(opts.path) != 0) {
               return;
             }
-            const writer = opts.getWriter ? opts.getWriter({path: extractPath}) : Writer({ path: extractPath });
-
-            return new Promise(function(resolve, reject) {
-              entry.stream(opts.password)
-                .on('error', reject)
-                .pipe(writer)
-                .on('close', resolve)
-                .on('error', reject);
+
+            return fs.ensureFile(extractPath).then(() => {
```

Comment:
This code is not efficient, as it will create an empty file and then subsequently overwrite it with the createWriteStream.   It's better to just focus on the directory, not the file.